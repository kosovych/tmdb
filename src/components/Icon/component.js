import React from 'react'
import PropTypes from 'prop-types'

const Icon = ({ as: Component, ...restProps }) => (
  <Component {...restProps} />
)

Icon.propTypes = {
  as: PropTypes.elementType.isRequired
}

export default Icon
