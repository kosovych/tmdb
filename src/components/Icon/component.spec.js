import { shallow } from 'enzyme'
import React from 'react'
import Icon from './component'

describe('Icon component', () => {
  const TestComponent = () => <div>Test</div>
  const defaultProps = {
    as: TestComponent
  }

  it('matches snapshot', () => {
    const wrapper = shallow(<Icon {...defaultProps} />)
    expect(wrapper).toMatchSnapshot()
  })
})
