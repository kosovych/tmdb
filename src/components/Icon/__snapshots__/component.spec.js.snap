// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component matches snapshot 1`] = `<TestComponent />`;
